compiler/src/java_plugin/cpp/java_generator.cpp

```diff
@@ -395,10 +407,37 @@ static void PrintStub(
             "    $input_type$ request)");
         break;
     }
-    if (impl) {
-      // Method body for client impls
-      p->Print(" {\n");
-      p->Indent();
+
+    if (!(abstract || impl)) {
+      // Interface method - there will be no body, close method.
+      p->Print(";\n");
+      continue;
+    }
+
+    // Method body for client impls
+    p->Print(" {\n");
+    p->Indent();
+
+    if (abstract) {
+      switch (call_type) {
+        case BLOCKING_CALL:
+          p->Print(
+              *vars,
+              "unimplementedStreamResponse(responseObserver);\n");
```

Comment:
This is C++, and if we follow the Google style guide it says [don't use exceptions](http://google.github.io/styleguide/cppguide.html#Exceptions).